### PR TITLE
Normalize package names

### PR DESCRIPTION
**Pull request by cdce8p**
Noticed that some wheels are always build and uploaded even though the current version already exists on https://wheels.home-assistant.io/musllinux/. Notably `charset-normalizer` and `pymicro-vad`. The reason for it is that the package name in the constraints file doesn't match the wheel name spelling => `-` vs `_`.

```
# package_constraints.txt
charset-normalizer==3.4.0
pymicro-vad==1.0.1

# wheel index
charset_normalizer-3.4.0-py3-none-any.whl
pymicro_vad-1.0.1-cp312-cp312-musllinux_1_2_x86_64.whl
```

This PR uses `packaging` to normalize the package names and prevent duplicate wheel builds and uploads.

https://github.com/home-assistant/core/actions/runs/11683262865/job/32532124722#step:12:2598

### CODE REVIEW

**Review by edenhaus — APPROVED** [review]
Thanks @cdce8p :+1: 
LGFM but I want a second review from @frenck

**Review by frenck — APPROVED** [review]
Nice! 🎉 

../Frenck

**Pull request merged**
